extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1183,46 +1140,186 @@ private JsonObject getVideoSecondaryInfoRenderer() throws ParsingException {
         return videoSecondaryInfoRenderer;
     }
 
+    @FunctionalInterface
+    private interface StreamBuilderHelper<T extends Stream> {
+        @Nonnull
+        T buildStream(ItagInfo itagInfo);
+    }
```

Comment:
I think you can just replace this with `Function<ItagInfo, T>` everywhere, and avoid defining a new interface. Do this only if you think it is simpler to understand, otherwise ignore my suggestion.

Reply:
I did this, since it also reduced code size